lib/purl/composer.ex

```diff
@@ -46,6 +46,11 @@ defmodule Purl.Composer do
     end)
   end
 
+  @spec encode_name(name :: Purl.name()) :: [String.t()]
+  defp encode_name(name) do
+    URI.encode(name, &(&1 != ?@ and URI.char_unescaped?(&1)))
```

Comment:
Is that correct?

https://github.com/package-url/purl-spec/blob/main/PURL-SPECIFICATION.rst#character-encoding

> the '@' version separator must be encoded as %40 elsewhere

Reply:
I took this `URI.encode` from `encode_namespace`. The condition here is whether the character is not encoded, so I think it's correct to say that `@` is not not encoded.

Reply:
Ups, I misread the Elixir `URI` docs. Makes sense